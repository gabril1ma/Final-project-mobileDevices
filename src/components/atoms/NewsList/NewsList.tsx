import React from "react";
import { ScrollView, StyleSheet, Text, View } from "react-native";
import NewsDTO from "../../../DTO/NewsDTO";
import NewsItem from "../NewsItems/newsItems";

const NewsList = ({ news }) => {
  return (
    <View style={styles.container}>
      <ScrollView>
        {news?.map((item: NewsDTO) => (
          <NewsItem
            key={item.metatags.og_title}
            broadcaster={item.metatags.og_site_name}
            imageUrl={item.metatags.og_image ?? item.metatags.twitter_image}
            link={item.link}
            ogTitle={item?.metatags?.og_title}
          />
        ))}

        {news.length === 0 && (
          <Text style={styles.noNewsText}>Não foram encontradas notícias</Text>
        )}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    // width: "60%",
  },
  noNewsText: {
    textAlign: "center",
    color: "#C4001F",
    fontSize: 16,
    marginTop: 20,
  },
});

export default NewsList;
