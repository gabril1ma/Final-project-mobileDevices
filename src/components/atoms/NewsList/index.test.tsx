import { create, ReactTestRendererJSON } from "react-test-renderer";
import NewsList from "./NewsList";

export type TTreeRenderer = ReactTestRendererJSON | ReactTestRendererJSON[];

const MOCK_TEXT = "Jest test";
const mockFunction = () => jest.fn();

describe("<NewsList />", () => {
  let tree: TTreeRenderer;
  let testTree: ReactTestRendererJSON;

  beforeEach(() => {
    tree = create(<NewsList />).toJSON();

    testTree = tree[0] ?? tree;
  });

  it("should render component", () => {
    expect(testTree).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(testTree?.children?.length).toBe(3);
  });
});
