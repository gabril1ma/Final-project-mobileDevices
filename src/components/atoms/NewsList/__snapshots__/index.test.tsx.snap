// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsList /> should render component 1`] = `
<View
  style={
    {
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://via.placeholder.com/50",
        }
      }
      style={
        {
          "borderRadius": 25,
          "height": 50,
          "marginRight": 10,
          "width": 50,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      News Title 1
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://via.placeholder.com/50",
        }
      }
      style={
        {
          "borderRadius": 25,
          "height": 50,
          "marginRight": 10,
          "width": 50,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      News Title 2
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://via.placeholder.com/50",
        }
      }
      style={
        {
          "borderRadius": 25,
          "height": 50,
          "marginRight": 10,
          "width": 50,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      News Title 3
    </Text>
  </View>
</View>
`;
